File: easy/0575_distribute_candies/solution.cpp
#include <unordered_set>
#include <vector>

class Solution {
public:
  int distributeCandies(std::vector<int> &candyType) {
    return std::min(
        static_cast<int>(
            std::unordered_set<int>(candyType.begin(), candyType.end()).size()),
        static_cast<int>(candyType.size()) / 2);
  }
};

File: easy/0575_distribute_candies/Solution.java
import java.util.Arrays;
import java.util.stream.Collectors;

class Solution {
  public int distributeCandies(int[] candyType) {
    return Math.min(
        Arrays.stream(candyType).boxed().collect(Collectors.toSet()).size(), candyType.length);
  }
}
